staging/src/k8s.io/cloud-provider/app/controllermanager.go

```diff
@@ -320,48 +320,65 @@ var ControllersDisabledByDefault = sets.NewString()
 
 // ConstructControllerInitializers is a private map of named controller groups (you can start more than one in an init func)
 // paired to their InitFunc.  This allows for structured downstream composition and subdivision.
-func ConstructControllerInitializers(initFuncConstructors map[string]InitFuncConstructor, completedConfig *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface) map[string]InitFunc {
+func ConstructControllerInitializers(controllerInitializerConstructors map[string]ControllerInitializerConstructor, completedConfig *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface) map[string]InitFunc {
 	controllers := map[string]InitFunc{}
-	for name, constructor := range initFuncConstructors {
-		controllers[name] = constructor(completedConfig, cloud)
+	for name, constructor := range controllerInitializerConstructors {
+		controllers[name] = constructor.Constructor(constructor.ClientName, completedConfig, cloud)
```

Comment:
I would just pass the constructor here.

Reply:
Same comment as above, I am passing a `constructor.InitContext` instead of the constructor because I think it slightly more readable but still future-proofs the signature.